sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/Util.java

```diff
@@ -371,9 +371,11 @@ public static URI convertNamespaceToEndPointURI(String namespaceName) {
     }
 
     public static ClientSettings getClientSettingsFromConnectionStringBuilder(ConnectionStringBuilder builder) {
-        TokenProvider tokenProvider;
-        if (builder.getAuthentication() != null && builder.getAuthentication().equalsIgnoreCase(ConnectionStringBuilder.MANAGED_IDENTITY_AUTHENTICATION)) {
-            tokenProvider = new ManagedIdentityTokenProvider();
+        TokenProvider tokenProvider = null;
+        if (builder.getAuthentication() != null) {
+            if (builder.getAuthentication().replace(" ", "").equalsIgnoreCase(ConnectionStringBuilder.MANAGED_IDENTITY_AUTHENTICATION.replace(" ", ""))) {
```

Comment:
>replace(" ", "") [](start = 44, length = 16)

Why are you doing? You shouldn't replace spaces inside a string, but only at the ends of it. Use string.trim() to remove begining and ending spaces.